noop/noop_sink.go

```diff
@@ -0,0 +1,41 @@
+package noop
+
+import (
+	"context"
+
+	"github.com/uw-labs/substrate"
+)
+
+// AsyncMessageSink does no operations on publish messages calls
+type AsyncMessageSink struct{}
+
+// NewAsyncMessageSink returns a pointer to a new AsyncMessageSink
+func NewAsyncMessageSink() (substrate.AsyncMessageSink, error) {
+	return &AsyncMessageSink{}, nil
+}
+
+// PublishMessages implements message publishing by only acknowledging the messages
+func (ams *AsyncMessageSink) PublishMessages(ctx context.Context, acks chan<- substrate.Message, messages <-chan substrate.Message) (rerr error) {
```

Comment:
You don't need the named return here. The functions also don't need to be defined on a pointer to the struct.